### PR TITLE
Add angle setter/getter to Rectangle

### PR DESCRIPTION
**Pull request by dstansby**
This adds an `angle` property, such that the `Rectangle` is labelled stale when it is updated. All the other properties use `set_` and `get_`, but to maintain backwards compatibility I've used a property here since `Rectangle.angle` already existed.

### CODE REVIEW

**Comment by timhoffm** [comment]
I propose to add `set_` and `get_` instead, but leave the storage in the public `Rectangle.angle`.

This is a consistency issue. We don't have attributes do fancy stuff (and thus have litte public attributes anyway). Defining `set_` enables the use with standard mechanisms like `rect.set(width=2, angle=30)`.

We don't need staling on `.angle` assignment. This has not been working and thus we don't need that for backward compatibility.

We may additionally deprecate `.angle`, but I'm undecided if that's worth it.

**Review by anntzer — APPROVED** [review]

File: lib/matplotlib/patches.py [hunk 1]
```diff
@@ -799,6 +799,10 @@ def get_height(self):
         """Return the height of the rectangle."""
         return self._height
 
+    def get_angle(self):
+        """Get the rotation angle."""
```
Comment (timhoffm) [inline lib/matplotlib/patches.py:803]:
```suggestion
        """Get the rotation angle in degrees."""
```

File: lib/matplotlib/patches.py [hunk 2]
```diff
@@ -809,6 +813,11 @@ def set_y(self, y):
         self._y0 = y
         self.stale = True
 
+    def set_angle(self, angle):
+        """Set the rotation angle."""
```
Comment (timhoffm) [inline lib/matplotlib/patches.py:817]:
```suggestion
        """
        Set the rotation angle in degrees.
        
        The rotation is performed anti-clockwise around *xy*.
        """
```

**Comment by anntzer** [comment]
My approval still stands, although you made flake8 unhappy :)

**Comment by jklymak** [comment]
Please add tests for the new setters and getters!

**Review by timhoffm — APPROVED** [review]
I don't think a test is strictly necessary for this, but of course @jklymak is right. Leaving open in case you want to add a test. If not, I'd also merge without.

**Comment by dstansby** [comment]
Test added. It's a simple figure comparison to make sure that setting the rotation after patch creation does the same thing as setting it in the constructor, with some additional tests for `get_angle()`.

**Pull request merged**